### ISSUE TITLE
quarkus-oidc ignores jvm proxy configuration and clashes with oidc-client when both proxies are set

### ISSUE THREAD

**Issue by jorgebsa** [quarkusio/quarkus#40469]
### Describe the bug

This is going to be tough to create a reliable reproducer, but I will provide as much context as I can right now. Let’s start by describing the application and the environment it runs on:

1) this application is built with quarkus 3.10 and Java 21
2) it is packaged in a Docker image built with jvm mode, with jvm http(s) proxy properties and custom ca certificates installed on container level. Base image is ubi8/openjdk-21
3) it is deployed to a kubernetes cluster of a global enterprise, which I don’t have access to (can’t access pods, can’t run debug mode)
4) it uses Microsoft entra as OIDC provider
5) it uses quarkus-rest-client for interaction with other APIs 
6) when deployed on the kubernetes cluster, all outgoing requests fail unless they go through a proxy server

The first few times the application was deployed, it was failing to startup, with the error message being that the OIDC server was not available at the auth-server-url location. Deploying it with the exact same configuration on any other environment that had no proxy constraint was successful, so it was clear the proxy was the issue. 

After deep reading of the documentation and debugging of OIDC classes, it became clear that quarkus-oidc does not honor jvm proxy properties, unlike quarkus-rest-client. When considering consistency of platform extensions, this seemed weird, but since it was not mentioned that should honor jvm configs on documentation, it is not really a bug. Once I figured that out, I’ve added both env vars QUARKUS_OIDC_PROXY_ADDRESS and QUARKUS_OIDC_PROXY_PORT which fixed the OIDC issues. Application started fine, and all test clients were working fine with the proxy just with JVM configs. 

Now comes the really weird part. A new requirements came in to use Microsoft Graph API to query some user data. Since all configuration was in place for the Microsoft Entra tenant, I’ve simply added the necessary quarkus-oidc-client and quarkus-rest-client-oidc-filter, and created a simple rest client for Graph with a single method for the call. Client was annotated with @RegisterRestClient(configKey=“graph”) and @OidcClientFilter(“graph”). The “graph” OIDC client configuration properties were mapped from the OIDC  tenant directly on application.yml with ${quarkus.oidc.auth-server-url} and such. 

Once the changes were ready, I’ve built it on CI and deployed to the cluster. To my surprise, I was getting again the same error that OIDC server was not available at the tenants location, even though none of the OIDC tenant’s configs were changed. I realized I had not added the oidc-client proxy properties, so i went back to the helm template and added both proxy address and port to the env vars for the graph client. Deployed again, same issue. Tried again with oidc-client global properties this time, no luck. Removed all graph client code and properties (not rolling back, just commenting out) rebuilt and deployed, everything works again.

I’ve typed all of this from phone, so it won’t be that easy to follow, and even harder for me to create a reproducer, as it would require an actual proxy setup for fidelity. But hopefully, with these details, someone can maybe point to a possible solution or something else that I could do to debug and provide more info.

### Expected behavior

OIDC and OIDC-CLIENT extensions should not interfere  on each other’s proxy configuration, if they share the same auth-server-url. 

For platform consistency, it would be good if OIDC used global level jvm proxy configuration, just like quarkus-rest-client

### Actual behavior

Setting just oidc proxy properties works, setting both oidc and oidc-client proxy properties for the same auth-server-url breaks things

### How to Reproduce?

Create an environment where all http calls must be proxied or else fail
Add quarkus-oidc and quarkus-oidc-client
Configure them with same auth-server-url and client details
Start the application without proxy properties: fail
Start the application with proxy properties: fail
Start the application with only oidc (remove client) and proxy properties: works

### Output of `java -version`

openjdk version “21.0.2” 2024-01-16 LTS
OpenJDK Runtime Environment (Red_Hat-21.0.2+13-LTS)
OpenJDK 64-Bit Server VM (Red_Hat-21.0.2.0.13-1) (build 21.0.2+13-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.10.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.5 (Red Hat 3.8.5-4)
Maven home: /usr/share/maven
Java version: 21.0.2, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-21-openjdk-21.0.2.0.13-1.el8.x86_64
Default locale: en, platform encoding: UTF-8

**Comment by sberyozkin** [comment]
Thanks @jorgebsa for the detailed description.

As far as `quarkus-oidc` and proxy is concerned, you mentioned that

>I’ve added both env vars QUARKUS_OIDC_PROXY_ADDRESS and QUARKUS_OIDC_PROXY_PORT which fixed the OIDC issues.

How did you set `quarkus-oidc-client` proxy properties with env vars ? As far as I recall due to the use of `-` in its configuration namespace, `quarkus.oidc-client`, one needs to declare it in properties first, for example `quarkus-oidc-client.proxy.host=` and then provide an env var like `QUARKUS_OIDC_CLIENT_PROXY_HOST`, though try with the env var only first.

I'm not sure about using JVM properties, as the requirement for OIDC can be totally different to other parts of the platform that need these proxy properties. For example,  REST client needs to use proxy to access ServiceA, but OIDC uses Google to authenticate users so it can't use JVM proxy settings...

**Comment by sberyozkin** [comment]
Please ignore my comments about using Google, as it is likely all the outbound calls can be managed by the same proxy properties. But in any case, I propose that we resolve the main issue where you can't use both OIDC client and OIDC with configured proxy properties and then you open an enhancement request for OIDC and OIDC client to check JVM proxy properties, thanks

**Comment by jorgebsa** [comment]
Thanks for the feedback Sergey, I can’t test it right now, but what you
mention about the namespace and env vars namespace makes sense. I will try
with the empty property being overwritten by the env and come back to you
as soon as I can. Once that’s settled, I will proceed with your suggestion
of opening an enhancement request. Thank you

**Comment by sberyozkin** [comment]
@jorgebsa Sure, please test when you can,  in general,  these 2 extensions are considered to be totally independent, as quarkus-oidc is about securing the endpoint,  and quarkus-oidc-client is only about acquring tokens and can be done with or without quarkus-oidc, their scopes can be totally different

**Comment by jorgebsa** [comment]
@sberyozkin so the issue indeed seems to be coming from the env vars setup. As explained, I had added `QUARKUS_OIDC_PROXY_HOST` and `QUARKUS_OIDC_PROXY_PORT`, which solved the problem for the `oidc` tenant. Then, I had to setup the `oidc-client` configuration, let's call it `abc-api`.

My initial attempt was to add the following environment variables to the helm template: `QUARKUS_OIDC_CLIENT__ABC_CLIENT__PROXY_PORT` and `QUARKUS_OIDC_CLIENT__ABC_CLIENT__PROXY_PORT`. Once deployed, application kept failing to start claiming that the OIDC server was unavailable, as the request wasn't proxied.

My second attempt was to keep those properties and add to the prod profile on `application.yml` the following properties with empty values: `quarkus.oidc-client."abc-client".proxy.host`, `quarkus.oidc-client."abc-client".proxy.port`. It still kept failing. 

I figured I must have missed something very basic in there, but given tiredness I went for the third attempt with something that would have to work no matter how tired I was, So, this time I went to the prod profile and mapped all oidc and oidc-client proxy properties to the env vars `HTTP_PROXY_HOST` and `HTTP_PROXY_PORT`, then went back to helm to remove all quarkus oidc and oidc-client proxy env var and added these 2 new envs. Deployed, and it all worked. 

So, it does indeed seem I was missing something on how I configured the application with env vars in this case. If you spot any evident problem, let me know. Other than that, I guess this ticket can be closed, I will proceed with the proposal for enhancement of oidc and oidc-client to honor jvm proxy properties.
